raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -787,9 +800,8 @@ private FetchQuorumRecordsResponseData tryCompleteFetchQuorumRecordsRequest(
             return buildEmptyFetchQuorumRecordsResponse(Errors.NONE, highWatermark)
                 .setNextFetchOffset(nextOffsetAndEpoch.offset)
                 .setNextFetchOffsetEpoch(nextOffsetAndEpoch.epoch);
-        } else if (quorum.isVoter(replicaId)) {
-            updateReplicaEndOffset(state, replicaId, fetchOffset);
         }
+        updateReplicaEndOffset(state, replicaId, fetchOffset);
```

Comment:
We probably need to check if `replicaId` is greater than 0. We will probably allow consumer fetches which will have a replicaId less than 0.